### ISSUE TITLE
ensure_comprehensions behavior changes with assignment

### ISSUE THREAD

**Issue by terite** [clutchski/coffeelint#465]
This is valid to coffeelint

```coffee
bar(b for b in baz)
```

This triggers "Comprehensions must have parentheses around them"
```coffee
foo = bar(b for b in baz)
```

Adding another set of parens fixes it up
```coffee
foo = bar((b for b in baz))
```

This seems like a straightforward bug to me. My belief is that `ensure_comprehensions` should not flag the second case.
